### DISCUSSION TITLE
Jackd server hang

### DISCUSSION THREAD

**Discussion by ArtemRotov** [jackaudio/jack2#921] [Q&A]
Hello. I'm not sure if this is the issue, so I'm not creating a issue yet. I have an application that uses jack2. If I interrupt the application through the debugger and `jack_client_close` is not called, then the jack server remains in a hung state. I can no longer run my application because it hangs on the `jack_client_open`. I have to do `pkill -ABRT jackd` to remove the process. Only after that I can start the application again. If closed correctly, then there is no such problem. Can you please tell me if this is the expected behaviour?

**Comment by falkTX** [comment]
Yes, the server will wait for the client side to close or process to exit, which both never do because it is manually stopped.
This could even be seen as the proper handling, as one might want to have the debug session lasting a long while. If jackd would kill the client by force, you would potentially lose the debug details.

**Reply by ArtemRotov** [reply]
Okay, I got it.
Is there a way inside the program to check that my previous server is already running and work with it? Or forcibly kill the process programmatically, without using the capabilities of the operating system directly?
